src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs

```diff
@@ -328,10 +328,18 @@ pub open spec fn every_delete_request_from_vrs_has_rv_precondition_that_is_less_
         }
     }
 }
+
+pub open spec fn vrs_does_not_have_deletion_timestamp (
```

Comment:
You should separate the two parts of this predicate. The first line (talking about `s.resources`) and the second line (talking about `s.ongoing_reconciles`) should be in separate predicates, since we can assume the first holds in the definition of ESR, but we need to prove that the second holds.